In [ ]:
from googleapiclient.discovery import build

API_KEY = 'AIzaSyBfhWPJXJ5qrAuWx_5JoNlKYHavGWB08yk'
VIDEO_ID = 'gmp2tS2joaA'

youtube = build('youtube', 'v3', developerKey=API_KEY)

request = youtube.commentThreads().list(
    part='snippet',
    videoId=VIDEO_ID,
    maxResults=5,
    order='relevance'
)
response = request.execute()

for item in response['items']:
    comment = item['snippet']['topLevelComment']['snippet']['textDisplay']
    print(comment)
    print('---')

In [ ]:
from googleapiclient.discovery import build
import csv
import time

API_KEY = 'AIzaSyBfhWPJXJ5qrAuWx_5JoNlKYHavGWB08yk'

videos = [
    ('campusx', 'gmp2tS2joaA'),
    ('campusx', 'fbKz7N92mhQ'),
    ('campusx', 'ZftI2fEz0Fw'),
    ('campusx', '3oOipgCbLIk'),
    ('campusx', 'dr7z7a_8lQw'),
    ('dhruv_rathee', 'fFMcolWgD3w'),
    ('dhruv_rathee', 'xlRBdUUJ8yc'),
    ('dhruv_rathee', 'oeH0fOSJuX8'),
    ('nitish_rajput', 'acunHsQcHS0'),
    ('nitish_rajput', 'nWc3c1Lvp1Q'),
    ('t_series', 'BBAyRBTfsOU'),
    ('t_series', 'nCD2hj6zJEc'),
    ('t_series', 'KhnVcAC5bIM'),
    ('old_song', 'CWfCp96-yck'),
    ('kapil_sharma', 'crAbfsVgLvY'),
    ('comedy', 'jh0gFT9Tgcs'),
    ('short', 'XPx_U-Gw5II'),
]

youtube = build('youtube', 'v3', developerKey=API_KEY)
all_comments = []

for i, (source, video_id) in enumerate(videos):
    order = 'relevance' if i % 2 == 0 else 'time'
    try:
        request = youtube.commentThreads().list(
            part='snippet',
            videoId=video_id,
            maxResults=25,
            order=order,
            textFormat='plainText'
        )
        response = request.execute()
        for item in response['items']:
            c = item['snippet']['topLevelComment']['snippet']
            all_comments.append({
                'source': source,
                'video_id': video_id,
                'order_used': order,
                'text': c['textDisplay'],
                'like_count': c['likeCount']
            })
        print(f"{source} ({video_id}): fetched {len(response['items'])} comments")
    except Exception as e:
        print(f"{source} ({video_id}): FAILED — {e}")
    time.sleep(0.5)

with open('yt_comments_raw.csv', 'w', newline='', encoding='utf-8') as f:
    writer = csv.DictWriter(f, fieldnames=['source', 'video_id', 'order_used', 'text', 'like_count'])
    writer.writeheader()
    writer.writerows(all_comments)

print(f"\nTotal comments fetched: {len(all_comments)}")

In [ ]:
import pandas as pd

df = pd.read_csv('yt_comments_raw.csv')
print(f"Before filtering: {len(df)}")

# Drop exact duplicate comments (same person quoting themselves, bot spam, etc.)
df = df.drop_duplicates(subset='text')
print(f"After dedup: {len(df)}")

# Drop very short comments (single word, pure emoji, "🔥🔥🔥") — adjust threshold as needed
df['word_count'] = df['text'].str.split().str.len()
df = df[df['word_count'] >= 2]
print(f"After min-length filter: {len(df)}")

# Drop obvious spam patterns
spam_keywords = ['subscribe my channel', 'check out my channel', 't.me/', 'watch my video']
df = df[~df['text'].str.lower().str.contains('|'.join(spam_keywords), na=False)]
print(f"After spam filter: {len(df)}")

# Optional: cap very long comments (essays, unrelated to sentiment task)
df = df[df['word_count'] <= 60]
print(f"After max-length filter: {len(df)}")

df.to_csv('yt_comments_filtered.csv', index=False)

In [ ]:
df = pd.read_csv("yt_comments_filtered.csv")
df.head()

In [ ]:
import pandas as pd

df = pd.read_csv('yt_comments_filtered.csv')

# Roughly 3 comments per video (17 videos × 3 ≈ 51, close enough to 50)
pilot = df.groupby('video_id', group_keys=False).apply(
    lambda x: x.sample(min(3, len(x)), random_state=42)
)

pilot = pilot.reset_index(drop=True)
pilot['manual_label'] = ''  # empty column for you to fill in by hand

pilot.to_csv('pilot_50.csv', index=False)
print(f"Pilot set size: {len(pilot)}")

In [4]:
df = pd.read_csv('pilot_50_labeled.csv')  
df.columns

Index(['source', 'video_id', 'order_used', 'manual_label', 'text',
       'like_count', 'word_count', 'sentiment_label'],
      dtype='object')

In [5]:
df.rename(columns={"sentiment_label" : "llm_label"},inplace=True)

In [6]:
df.columns

Index(['source', 'video_id', 'order_used', 'manual_label', 'text',
       'like_count', 'word_count', 'llm_label'],
      dtype='object')

In [7]:
import pandas as pd

#df = pd.read_csv('pilot_50_labeled.csv')  # replace with your actual filename



# Quick sanity check first — make sure both columns loaded properly
print(df[['manual_label', 'llm_label']].head(10))
print(df[['manual_label', 'llm_label']].dtypes)

# Simple accuracy: % of rows where both labels match exactly
agreement = (df['manual_label'] == df['llm_label']).mean()
print(f"Simple agreement: {agreement:.2%}")

   manual_label  llm_label
0             0          0
1             0          0
2             1          1
3             0          0
4             0          0
5             1          1
6             0          1
7             1          1
8             0          0
9             1          1
manual_label    int64
llm_label       int64
dtype: object
Simple agreement: 89.13%


In [8]:
from sklearn.metrics import cohen_kappa_score

kappa = cohen_kappa_score(df['manual_label'], df['llm_label'])
print(f"Cohen's kappa: {kappa:.3f}")

Cohen's kappa: 0.822


Rough interpretation guide for the kappa number:

- < 0.4 — weak agreement, prompt needs work
- 0.4 - 0.6 — moderate, borderline usable
- 0.6 - 0.8 — good, generally trustworthy
- "> 0.8" — excellent, very strong agreement

In [9]:
disagreements = df[df['manual_label'] != df['llm_label']]
print(f"\n{len(disagreements)} disagreements out of {len(df)}:\n")
print(disagreements[['text', 'manual_label', 'llm_label']].to_string())


5 disagreements out of 46:

                                                                                                                                                                                                                                                                                                                                                                        text  manual_label  llm_label
6   🎧 Experience the magic of pure romance…\nPresenting the full audio of the evergreen love anthem Teri Ore from Singh Is Kinng! ❤🎶\n\n🎤 Vocals: Rahat Fateh Ali Khan & Shreya Ghoshal\n🎼 Music: Pritam | 🎬 Starring: Akshay Kumar & Katrina Kaif\n\n💬 Let us know where Teri Ore takes you...\n#TeriOre #FullAudio #SinghIsKinng #RahatFatehAliKhan #ShreyaGhoshal #Pritam             0          1
30                                                                                                                                                                                             

In [10]:
df = pd.read_csv("yt_comments_filtered_labeled.csv")
df.head()

,source,video_id,order_used,text,like_count,word_count,llm_label
0,campusx,gmp2tS2joaA,relevance,Happy Birthday Nitish Sir. One of the best tea...,15,13,1
1,campusx,gmp2tS2joaA,relevance,Absolutely love this series. Can you please ma...,13,17,1
2,campusx,gmp2tS2joaA,relevance,if we have two features then one thing we can ...,12,32,0
3,campusx,gmp2tS2joaA,relevance,"Happy birthday sir.I love your commitments, di...",7,9,1
4,campusx,gmp2tS2joaA,relevance,This is the best explanation i have ever seen ...,8,57,-1


In [11]:
df.to_csv("evaluation_dataset.csv",index= False)

In [12]:
df.head()

,source,video_id,order_used,text,like_count,word_count,llm_label
0,campusx,gmp2tS2joaA,relevance,Happy Birthday Nitish Sir. One of the best tea...,15,13,1
1,campusx,gmp2tS2joaA,relevance,Absolutely love this series. Can you please ma...,13,17,1
2,campusx,gmp2tS2joaA,relevance,if we have two features then one thing we can ...,12,32,0
3,campusx,gmp2tS2joaA,relevance,"Happy birthday sir.I love your commitments, di...",7,9,1
4,campusx,gmp2tS2joaA,relevance,This is the best explanation i have ever seen ...,8,57,-1


In [14]:
df = df[["text", "llm_label"]]

In [15]:
# Define the preprocessing function
def preprocess_comment(comment):
    # Convert to lowercase
    comment = comment.lower()

    # Remove trailing and leading whitespaces
    comment = comment.strip()

    # Remove newline characters
    comment = re.sub(r'\n', ' ', comment)

    # Remove non-alphanumeric characters, except punctuation
    comment = re.sub(r'[^A-Za-z0-9\s!?.,]', '', comment)

    # Remove stopwords but retain important ones for sentiment analysis
    stop_words = set(stopwords.words('english')) - {'not', 'but', 'however', 'no', 'yet'}
    comment = ' '.join([word for word in comment.split() if word not in stop_words])

    # Lemmatize the words
    lemmatizer = WordNetLemmatizer()
    comment = ' '.join([lemmatizer.lemmatize(word) for word in comment.split()])

    return comment

In [18]:
df.shape

(374, 2)

In [22]:
import re
import nltk
from nltk.corpus import stopwords
from nltk.stem import WordNetLemmatizer

In [25]:
df["text"]=df["text"].apply(preprocess_comment)

In [26]:
df.isna().sum()

text         0
llm_label    0
dtype: int64

In [ ]:
import joblib
import pandas as pd
import spacy
from scipy.sparse import hstack, csr_matrix

# ── 1. Load the saved artifact ──────────────────────────────────
artifact = joblib.load('LightGBM_best_model.joblib') #changed the model

In [28]:
# ── 2. Load spaCy (needed to rebuild the custom features) ───────
nlp = spacy.load(artifact['spacy_model'])          # 'en_core_web_sm'
UNIVERSAL_POS = artifact['universal_pos']

In [29]:

# ── 3. Same feature extraction used during training ─────────────
def extract_custom_features(text):
    doc = nlp(text)
    word_list = [token.text for token in doc]

    comment_length    = len(text)
    word_count        = len(word_list)
    avg_word_length   = sum(len(w) for w in word_list) / word_count if word_count > 0 else 0
    unique_word_count = len(set(word_list))
    lexical_diversity = unique_word_count / word_count if word_count > 0 else 0
    pos_count         = len([token.pos_ for token in doc])

    pos_tags = [token.pos_ for token in doc]
    if word_count > 0:
        pos_proportion = {tag: pos_tags.count(tag) / word_count for tag in UNIVERSAL_POS}
    else:
        pos_proportion = {tag: 0 for tag in UNIVERSAL_POS}

    return {
        'comment_length': comment_length,
        'word_count': word_count,
        'avg_word_length': avg_word_length,
        'unique_word_count': unique_word_count,
        'lexical_diversity': lexical_diversity,
        'pos_count': pos_count,
        **pos_proportion
    }


In [ ]:
# ── 4. Predict ──────────────────────────────────────────────────
def predict_comments(comments):

    
    feats = pd.DataFrame([extract_custom_features(c) for c in comments])
    feats = feats.reindex(columns=artifact['custom_columns'], fill_value=0)

    X = hstack([
        artifact['tfidf'].transform(comments),
        csr_matrix(feats.values),
    ]).tocsr()

    return artifact['model'].predict(X)

In [31]:
def predict_comments(comments):
    comments = list(comments)          # accept Series or list safely

    feats = pd.DataFrame([extract_custom_features(c) for c in comments])
    feats = feats.reindex(columns=artifact['custom_columns'], fill_value=0)

    X = hstack([
        artifact['tfidf'].transform(comments),
        csr_matrix(feats.values),
    ]).tocsr()

    return artifact['model'].predict(X)

In [32]:
print(df["llm_label"].value_counts())
print("empty after preprocessing:", (df["text"].str.strip() == "").sum())

llm_label
 0    218
 1    138
-1     18
Name: count, dtype: int64
empty after preprocessing: 10


In [33]:
from sklearn.metrics import accuracy_score, classification_report, confusion_matrix

y_true = df["llm_label"].values
y_pred = predict_comments(df["text"])

print(f"Accuracy : {accuracy_score(y_true, y_pred):.4f}")
print()
print(classification_report(y_true, y_pred, digits=4))
print("Confusion matrix (rows = true, cols = predicted, order -1/0/1):")
print(confusion_matrix(y_true, y_pred, labels=[-1, 0, 1]))

Accuracy : 0.7246

              precision    recall  f1-score   support

          -1     0.3889    0.3889    0.3889        18
           0     0.7719    0.8073    0.7892       218
           1     0.6875    0.6377    0.6617       138

    accuracy                         0.7246       374
   macro avg     0.6161    0.6113    0.6133       374
weighted avg     0.7223    0.7246    0.7229       374

Confusion matrix (rows = true, cols = predicted, order -1/0/1):
[[  7   4   7]
 [  9 176  33]
 [  2  48  88]]


In [34]:
empty_rows = df[df['text'].str.strip() == ""]
print(empty_rows)

    text  llm_label
125               0
127               0
164               0
171               0
312               0
313               1
318               1
324               1
325               1
330               0


In [35]:
# assuming you have y_true, y_pred, and the original text aligned
mask = (y_true == 1) & (y_pred == 0)
misclassified = df[mask]['text']
print(misclassified.to_list())

['waiting this... thank sir.', 'power sir', 'thank sir.', 'thank sir', '2900', 'thank sir!', 'started today 12 march, thursday 12032026 comment back finishing', 'campusx going next big thing edtech. load respect sir', '13 may 2026', 'day 3 class 4 5 thank sir', 'leaving hope learn ml conceptually, channel blessing for. keep uploading videos.', 'think dhruv rathi need support jantar mantar protest', 'vote cjp', 'mein pakistan se hn ... mgr mera support youth cockroach party k sth h ... please request hai ... delhi student k liye dekhao', 'join ai bootcamp httpsacademy.dhruvrathee.comaibootcamp learn make ai product like apps games. valuable upskill get today era.', 'respect bharat tiwari...', 'sonam wangchuk respect button', 'bharat tiwari case please vote', 'video sir ratan tata ji', 'mandir chori ka adda', 'hoo', 'kalyan singh ji yogi ji', 'listen 330 repeat hit like thank u sab log like karne ki liye you? u tell plz', 'hey.... series please release nikhils solo song plzzz', 'anyone 2

In [36]:
df_eval = df[df['text'].str.strip() != ""].copy()
print(f"dropped {len(df) - len(df_eval)} empty rows, {len(df_eval)} remain")

y_true = df_eval['llm_label'].values
y_pred = predict_comments(df_eval['text'])

print(f"\nAccuracy: {accuracy_score(y_true, y_pred):.4f}")
print(classification_report(y_true, y_pred, digits=4))

dropped 10 empty rows, 364 remain

Accuracy: 0.7280
              precision    recall  f1-score   support

          -1     0.3889    0.3889    0.3889        18
           0     0.7798    0.8019    0.7907       212
           1     0.6875    0.6567    0.6718       134

    accuracy                         0.7280       364
   macro avg     0.6187    0.6158    0.6171       364
weighted avg     0.7265    0.7280    0.7270       364



In [37]:
# Reload the ORIGINAL text (your df has already been preprocessed)
orig = pd.read_csv("evaluation_dataset.csv")[["text", "llm_label"]].reset_index(drop=True)
proc = df.reset_index(drop=True)

eva = pd.DataFrame({
    "original": orig["text"],
    "clean":    proc["text"],
    "label":    proc["llm_label"],
})

eva["has_hindi"] = eva["original"].str.contains(r"[\u0900-\u097F]", na=False)
eva["is_empty"]  = eva["clean"].str.strip() == ""

print("total            :", len(eva))
print("contains Hindi   :", eva["has_hindi"].sum())
print("empty after prep :", eva["is_empty"].sum())
print()
print(pd.crosstab(eva["has_hindi"], eva["is_empty"]))

total            : 374
contains Hindi   : 18
empty after prep : 10

is_empty   False  True 
has_hindi              
False        356      0
True           8     10


In [38]:
sub = eva[(~eva["has_hindi"]) & (~eva["is_empty"])]
print(f"English-only, non-empty: {len(sub)} of {len(eva)} comments\n")

y_true = sub["label"].values
y_pred = predict_comments(sub["clean"])

print(f"Accuracy: {accuracy_score(y_true, y_pred):.4f}")
print(classification_report(y_true, y_pred, digits=4))

English-only, non-empty: 356 of 374 comments

Accuracy: 0.7303
              precision    recall  f1-score   support

          -1     0.3889    0.4118    0.4000        17
           0     0.7820    0.8010    0.7914       206
           1     0.6929    0.6617    0.6769       133

    accuracy                         0.7303       356
   macro avg     0.6213    0.6248    0.6228       356
weighted avg     0.7299    0.7303    0.7299       356



---

# Re-evaluation with Claude labels

The first pass used `llm_label` as ground truth, and it had visible problems — dates and
timestamps such as `"2900"` and `"13 may 2026"` were marked **positive**, and a lot of
clearly critical political/temple comments were marked **neutral**. That means part of the
72% score was the *labels* being wrong, not the model.

`claude_label` is a second, more careful labelling pass over the same comments.

The cells below:
1. compare the two label sets against each other
2. show what actually changed
3. re-score **the same model predictions** against each set

Because the predictions are identical in both cases, any difference in the score comes
purely from the labels.

In [40]:
claude_df = pd.read_csv("evaluation_dataset_with_claude_labels.csv")

print("rows   :", len(claude_df))
print("columns:", claude_df.columns.tolist())

print("\nllm_label distribution:")
print(claude_df["llm_label"].value_counts().sort_index())

print("\nclaude_label distribution:")
print(claude_df["claude_label"].value_counts().sort_index())

claude_df.head()

rows   : 358
columns: ['llm_label', 'text', 'claude_label']

llm_label distribution:
llm_label
-1      7
 0    213
 1    138
Name: count, dtype: int64

claude_label distribution:
claude_label
-1     52
 0    130
 1    176
Name: count, dtype: int64


,llm_label,text,claude_label
0,1,Happy Birthday Nitish Sir. One of the best tea...,1
1,1,Absolutely love this series. Can you please ma...,1
2,0,if we have two features then one thing we can ...,0
3,1,"Happy birthday sir.I love your commitments, di...",1
4,1,This is the best explanation i have ever seen ...,1


In [41]:
# How far apart are the two label sets?
changed = (claude_df["llm_label"] != claude_df["claude_label"]).sum()
print(f"changed: {changed} of {len(claude_df)}  ({changed/len(claude_df):.1%})")
print(f"Cohen's kappa (llm vs claude): {cohen_kappa_score(claude_df['llm_label'], claude_df['claude_label']):.3f}")

print("\nrows = llm_label, cols = claude_label, order -1/0/1")
print(confusion_matrix(claude_df["llm_label"], claude_df["claude_label"], labels=[-1, 0, 1]))

changed: 100 of 358  (27.9%)
Cohen's kappa (llm vs claude): 0.528

rows = llm_label, cols = claude_label, order -1/0/1
[[  7   0   0]
 [ 44 122  47]
 [  1   8 129]]


In [42]:
# What actually changed between the two label sets?
diff = claude_df[claude_df["llm_label"] != claude_df["claude_label"]]
print(f"{len(diff)} disagreements\n")

for _, r in diff.head(30).iterrows():
    print(f"llm={r['llm_label']:>2}  claude={r['claude_label']:>2}  |  {str(r['text'])[:95]}")

100 disagreements

llm= 1  claude= 0  |  for multiple features, i believe it will evaluate each feature separately and find the best spl
llm= 0  claude= 1  |  Hidden Pearl In YouTube 💚
llm= 0  claude= 1  |  This guy was ahead of time, 5 years ago, he was discussing how ML in important for future and I
llm= 0  claude= 1  |  The only teacher who gets the need and value of bird's eye view intros!!
llm= 1  claude= 0  |  I think dhruv rathi needs to support the jantar mantar protest🙏
llm= 1  claude= 0  |  Vote for CJP❤❤
llm= 0  claude= 1  |  Moral of the video : If Dhruv Rathee cannot solve this mystery, then no one can.✅
llm= 0  claude=-1  |  We want new not your type
llm= 0  claude=-1  |  एक व्हिडिओ गोदी मीडिया के नाम..
llm= 0  claude=-1  |  *Dharmendra Pradhan resign Button* 🔘 ✅
llm= 0  claude=-1  |  19:23 मुझसे भी लिख के ले लो, जिस दिन Mr. मोदी पावर से हटे, अडानी के शेयर भी धड़ाम से गिरेंगे 😂😂
llm= 0  claude=-1  |  Government on poor people :☠️☠️☠️
Government on rich people:🤡🤡🤡
llm= 0  

In [43]:
from sklearn.metrics import f1_score

# Preprocess with the SAME function used during training, then drop comments
# that get reduced to an empty string (they carry no signal).
ev = claude_df.copy()
ev["clean"] = ev["text"].astype(str).apply(preprocess_comment)
ev = ev[ev["clean"].str.strip() != ""].copy()
print(f"{len(ev)} non-empty comments (dropped {len(claude_df) - len(ev)})\n")

ev["pred"] = predict_comments(ev["clean"])

# Same predictions, scored against each set of labels
for name, truth in [("vs LLM labels", "llm_label"), ("vs Claude labels", "claude_label")]:
    acc = accuracy_score(ev[truth], ev["pred"])
    f1  = f1_score(ev[truth], ev["pred"], average="macro")
    print(f"{name:20s}  accuracy={acc:.4f}   macroF1={f1:.4f}")

348 non-empty comments (dropped 10)

vs LLM labels         accuracy=0.7529   macroF1=0.6374
vs Claude labels      accuracy=0.6236   macroF1=0.5308


In [44]:
# Full report against the Claude labels
print(classification_report(ev["claude_label"], ev["pred"], digits=4))

print("Confusion matrix (rows = true, cols = predicted, order -1/0/1)")
print(confusion_matrix(ev["claude_label"], ev["pred"], labels=[-1, 0, 1]))

              precision    recall  f1-score   support

          -1     0.5000    0.1800    0.2647        50
           0     0.5146    0.8413    0.6386       126
           1     0.8226    0.5930    0.6892       172

    accuracy                         0.6236       348
   macro avg     0.6124    0.5381    0.5308       348
weighted avg     0.6647    0.6236    0.6099       348

Confusion matrix (rows = true, cols = predicted, order -1/0/1)
[[  9  34   7]
 [  5 106  15]
 [  4  66 102]]


---

## Which label set is actually right?

Scoring the model against the two label sets gives very different answers:

| Scored against | Accuracy | Macro F1 |
|---|---|---|
| LLM labels | 0.7529 | 0.6374 |
| Claude labels | 0.6236 | 0.5308 |

The **model is identical** in both cases — only the answer key changed. So before drawing any
conclusion about the model, we need to know which key to trust.

The two label sets agree only moderately (kappa **0.528**), and they disagree in a very specific
way: of the 213 comments the LLM called *neutral*, Claude says 91 actually carry sentiment.
One of them is systematically wrong.

**The tiebreaker:** the 46 comments labelled **by hand**. Whichever label set agrees better with
those is the one to trust.

- Claude kappa **> 0.822** → Claude's labels are better; 0.62 is the honest score, and the model
  has a genuine bias toward predicting *neutral*.
- Claude kappa **< 0.822** → Claude is over-labelling sentiment; keep the original labels and the
  ~0.75 figure.

In [45]:
# Your 46 hand-labelled comments are the closest thing to ground truth here.
# Score BOTH label sets against them and see which one you actually agree with.
pilot = pd.read_csv("pilot_50_labeled.csv")
pilot = pilot.rename(columns={"sentiment_label": "llm_label"})[["text", "manual_label", "llm_label"]]
pilot = pilot.drop_duplicates(subset="text")

cl = claude_df[["text", "claude_label"]].drop_duplicates(subset="text")
m  = pilot.merge(cl, on="text", how="inner")

print(f"matched {len(m)} of {len(pilot)} hand-labelled comments\n")

print(f"YOU vs LLM     agreement: {(m['manual_label'] == m['llm_label']).mean():.1%}"
      f"   kappa: {cohen_kappa_score(m['manual_label'], m['llm_label']):.3f}")
print(f"YOU vs Claude  agreement: {(m['manual_label'] == m['claude_label']).mean():.1%}"
      f"   kappa: {cohen_kappa_score(m['manual_label'], m['claude_label']):.3f}")

matched 43 of 46 hand-labelled comments

YOU vs LLM     agreement: 90.7%   kappa: 0.842
YOU vs Claude  agreement: 97.7%   kappa: 0.960


In [46]:
# Where does Claude disagree with YOUR hand labels?
d = m[m["manual_label"] != m["claude_label"]]
print(f"{len(d)} disagreements with your labels:\n")
for _, r in d.iterrows():
    print(f"you={r['manual_label']:>2}  claude={r['claude_label']:>2}  |  {str(r['text'])[:90]}")

print("\n" + "-" * 70)

# And where the LLM disagreed with you, for comparison
d2 = m[m["manual_label"] != m["llm_label"]]
print(f"\n{len(d2)} disagreements between you and the LLM:\n")
for _, r in d2.iterrows():
    print(f"you={r['manual_label']:>2}  llm={r['llm_label']:>2}  |  {str(r['text'])[:90]}")

1 disagreements with your labels:

you= 0  claude=-1  |  भाई भगवान ऊपर कैसे चढ़ाते हुए कॉलेज से गरीबों पर कुछ करवा दो पिया दो😢😢

----------------------------------------------------------------------

4 disagreements between you and the LLM:

you= 0  llm= 1  |  okay, this is why the n_estimators has to be larger when the learning rate is smaller.
you= 0  llm=-1  |  Abhi thik kardete Hun 😂wtf
you= 1  llm= 0  |  We miss those childhood day 😢
you= 0  llm=-1  |  भाई भगवान ऊपर कैसे चढ़ाते हुए कॉलेज से गरीबों पर कुछ करवा दो पिया दो😢😢
